### DISCUSSION TITLE
"Functionality is not available" error when using checkpoint

### DISCUSSION THREAD

**Discussion by YYang-farm** [idaholab/moose#16529] [Q&A General] [answered]
Dear MOOSE team
I recently tried to install moose on our HPC referring to the procedure from the installation pages. However, once I finished the compilation of libmesh and do some tests with "checkpoint=true" set, it returned the following error message

_ERROR: Functionality is not available.
Make sure LIBMESH_HAVE_XDR is defined at build time
The XDR interface is not available in this installation_

I simply ran the following commends while compiling the libmesh
_scripts/update_and_rebuild_libmesh.sh --with-dof-id-bytes=8 --enable-vtk --with-vtk-include=$VTKINCLUDE_DIR --with-vtk-lib=$VTKLIB_DIR_

it seems that I should add some options to enable the XRD, which had never been set in the former procedure. 
Could anyone inform me of the solution to this issue? 
Thank you in advance

**Comment by YYang-farm** [comment]
Did anyone encounter the same error message as me?

**Comment by aeslaughter** [comment, marked as answer]
I imaging that there is some missing package on your machine that is causing libMesh to disable XDR support. If you look at text in during the configure, can you find the sections that mention XDR?

**Reply by YYang-farm** [reply]
Thank you very much for your reply. I saved the log of compiling the libMesh, and it indeed mentioned that "XDR is not supported."
/////
checking for built-in XDR support... no
checking for XDR support in /usr/include/tirpc... no
////
I compiled this on our newest machine with 96 avx512 cores and essences:  gcc/8.3.1, openmpi/4.0.4, cmake/3.18.4,
which is wield since I changed to another machine with different compiling essences (24 avx2 cores w/ gcc/8.3.0, openmpi/4.0.2, cmake/3.18.4), and at that time, I could successfully compile the libmesh with XDR. 

Is this issue occurred solely due to the differences of compelling packages, or is it also related to the usage of avx512 cores?
Thank you again, and look forward to your prompt reply.

**Reply by aeslaughter** [reply]
@milljm Do you know what could be missing that is causing XDR to be disabled?

**Reply by GiudGiud** [reply]
See this answer for future reference https://github.com/idaholab/moose/discussions/20479#discussioncomment-2311198
